### ISSUE TITLE
ng-click is funny behavior in v1.4.2

### ISSUE THREAD

**Issue by pyonnuka** [angular/angular.js#12353]
When I click on tag with ng-click,

```html
<button class="btn btn-warning" ng-click="updateGroup()" ng-disabled="updateGroup.$invalid">update</button>
````

get this error.
```text
TypeError: v2.updateGroup is not a function
    at fn (eval at <anonymous> 
```

Where "v2" come from?

**Comment by pkozlowski-opensource** [comment]
@pyonnuka please provide a minimal (but complete) reproduce scenario, see bug submission guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#submitting-an-issue

**Comment by gkalpak** [comment]
`v2` is coming from the parser (it is basically your current `scope`).
This error happens because you are trying to call `scope.updateGroup()`, but `scope.updateGroup` is not a function (it is defined, but it is not a function).

Since you seem to be using `updateGroup.$invalid`, I asume you might have form named `updateGroup`, in which case it would overwrite `scope.updateGroup` function already existing on the `scope`.

--
Bottom line, either change your function name or change your form name, so the latter does not overwrite the former.

**Comment by gkalpak** [comment]
In fact, it [used to throw][1] `TypeError: Cannot read property '0' of null` on v1.3.x, so I guess this is a nice improvement (the error makes much more sense now). 

[1]: http://codepen.io/ExpertSystem/pen/bdKzYM?editors=100

**Comment by lgalfaso** [comment]
As @gkalpak mention, the forgiving nature of angular expressions never included function calls on things that are defined but are not callable.

**Comment by Dman89** [comment]
@gkalpak I love you right now, it might be lust.

just to recap, the function has a name as another variable and is conflicting in $scope after being called

**Comment by olexy** [comment]
@gkalpak Thanks so much

**Comment by leosco** [comment]
This error usually happens when you're trying to call a scope variable as a function when it's not actually a function, i.e.
```
// html inside your controller element
<button ng-click="sayHello()">Hello</button>
// js in your controller
$scope.sayHello= 'hello';
```
It should be a simple fix, just track down where you're reassigning or misdeclaring `$scope.updateGroup`
